app/presenters/sufia/file_set_presenter.rb

```diff
@@ -6,6 +6,14 @@ class FileSetPresenter < ::CurationConcerns::FileSetPresenter
     delegate :depositor, :keyword, :date_created, :date_modified, :itemtype,
              to: :solr_document
 
+    def page_title
+      title.first
+    end
+
+    def link_name
+      current_ability.can?(:read, id) ? page_title : 'File'
+    end
+
```

Comment:
I'd merge this PR w/o integration testing, though, because, ugh, slow feature specs.